packages/jsii-diff/lib/type-analysis.ts

```diff
@@ -211,7 +211,7 @@ class TypePairs {
     if (!a.fqn || !b.fqn) { return false; } // Only for user-defined types
 
     const image = this.pairs.get(a.fqn);
-    return !!image && image.has(b.fqn);
+    return !!image?.has(b.fqn);
```

Comment:
why do we need the `!!` at the beginning here?

Reply:
Because I need to coerce this to a boolean context due to surrounding typing.